src/Microsoft.Performance.SDK.Runtime.NetCoreApp.Tests/Plugins/PluginsLoaderTests.cs

```diff
@@ -125,6 +125,48 @@ public static void Initialize(TestContext context)
             }
         }
         
+        [TestMethod]
+        [UnitTest]
+        public void NullDirectoryThrows()
+        {
+            (var loader, _) = Setup(true);
+            Assert.ThrowsException<ArgumentNullException>(() => loader.TryLoadPlugin(null, out _));
+        }
+
+        [TestMethod]
+        [UnitTest]
+        public void NullDirectoriesThrows()
+        {
+            (var loader, _) = Setup(true);
+            Assert.ThrowsException<ArgumentNullException>(() => loader.TryLoadPlugins(null, out _));
+        }
+        
+        [TestMethod]
+        [UnitTest]
+        public void NullDirectoryInDirectoriesThrows()
+        {
+            (var loader, _) = Setup(true);
+            Assert.ThrowsException<ArgumentNullException>(() => loader.TryLoadPlugins(new List<string>() { "foo", null }, out _));
+        }
+        
+        [TestMethod]
+        [UnitTest]
+        public void EmptyStringFailsToLoad()
+        {
+            (var loader, var consumer) = Setup(true);
+            var success = loader.TryLoadPlugin(string.Empty, out _);
+            Assert.IsFalse(success);
+        }
+        
+        [TestMethod]
+        [UnitTest]
+        public void EmptyDirectoriesLoads()
```

Comment:
IK it might be repetitive, but what about testing TryLoadPluginAsync

Reply:
Added. I left one commented out because I need to fix #222  for it to compile. Since I want to backport this fix to the 1.0 servicing branch, I'm going to fix that in a separate PR.